atlp-rwanda/e-commerce-mavericcks-bn
# [finishes 187354241]-implement-user-management-CRUD-operation

## Purpose

This pull request aims to implement CRUD (Create, Read, Update, Delete) operations for users. It includes changes made in the user controller, user router files, and migrations to add columns for roleId and Roles table in the database schema.

## Changes Made

- Updated the user controller to handle CRUD operations for users.
- Modified the user router to route requests to the appropriate controller methods.
- Added migration files to create columns for roleId in the database schema.

## Testing Instructions

1. Run the test suite to ensure all CRUD operations function as expected.
2. Manually test each endpoint using tools like Postman or through automated integration tests.
3. Verify that the database schema is updated correctly after running migrations.

## Related Issues

Reference any related issues or pull requests

## Checklist

Please review the following checklist and make sure all tasks are complete before submitting:
